department-of-veterans-affairs/abd-vro
# MCP-2347 - Removing "test" profile and use MockBipApiService only for tests.

Refactor code to remove "test" profile dependance for BipApiService, and prepare for using BIP mock service for local environment.

## What was the problem?
The current code uses MockBipApiService for BIP Api Service in local environment. We try to use a BIP mock service created in the application package as a service in a docker container, and simplify the BIP API service testing and configuration.

Associated tickets or Slack threads:

MCP-2347

## How does this fix it?[^1]

1. Refactor the code to use MockBipApiService for test only. 
2. Simplify IBipApiService instantiation. 
3. Minor code cleanup. 

## How to test this PR
- Step 1. Build the app. All tests should be passed.
- Step 2. Run in local environment, check automatedClaim endpoint. 
- Step 3. Set ENV to dev (export ENV=dev), and launch the app again (./gradlew :app:dockerComposeDown :app:dcPrune :app:dockerComposeUp). Repeat Step 2. A 500 response with a message containing "java.net.UnknownHostException: claims-uat.stage8.bip.va.gov: Name does not resolve" should be shown as the local environment doesn't have the connection to BIP uat.


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
